scripts/zkai_native_seq32_attention_mlp_predecommit_opening_policy_gate.py

```
@@ -0,0 +1,593 @@
+        write_outputs(args.write_json, args.write_tsv, payload)
+    else:
+        print(json.dumps(payload, indent=2, sort_keys=True))
+    return 0
+
+
+if __name__ == "__main__":
+    try:
+        raise SystemExit(main())
+    except PredecommitOpeningPolicyGateError as err:
+        print(f"error: {err}", file=sys.stderr)
+        raise SystemExit(2)
```

Ruff (0.15.13)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)